lantern_hnsw/scripts/test_updates.py

```diff
@@ -49,7 +49,8 @@ def __repr__(self):
         return self.version
 
 INCOMPATIBLE_VERSIONS = {
-    '16': [Version('0.0.4')]
+    '16': [Version('0.0.4')],
+    '17': [Version('0.3.0'), Version('0.3.1'), Version('0.3.2'), Version('0.3.3'), Version('0.3.4'), Version('0.4.0'), Version('0.4.1')],
```

Comment:
If we no longer have servers at an older `0.3.*` version that we need to upgrade from, you can also move old update scripts to [old_updates](https://github.com/lanterndata/lantern/tree/main/lantern_hnsw/sql/old_updates)

Reply:
there are still servers with 0.3.x versions, will move the update files to old_updates once all servers are upgraded